### PR TITLE
Streamlit app and first LLM agent iteration

### PR DESCRIPTION
**Pull request by nifets-ai**
Sorry I'm too lazy to write a description but let me know if you've got any questions.

A lot of it is ironing out the codebase to make it easier to use and clarifying the environment, observation space and action space to the LLM.

### CODE REVIEW

**Comment by Johnno1011** [comment]
is the failing check just normal primaite behaviour ?

**Comment by nifets-ai** [comment]
> is the failing check just normal primaite behaviour ?

yeah that happens.. not sure where from but chose to ignore it for now, later on we can have a deep dive on this

File: src/primaite/common/enums.py [hunk 1]
```diff
@@ -168,20 +168,39 @@ class FileSystemState(Enum):
...
+    FILE_SYSTEM = 4
+
+
 class NodeHardwareAction(Enum):
     """Node hardware action."""
 
     NONE = 0
-    ON = 1
-    OFF = 2
+    TURN_ON = 1
```
Comment (Johnno1011) [inline src/primaite/common/enums.py:185]: for my understanding - were all these just incorrect in the primaite codebase

Comment (nifets-ai) [reply src/primaite/common/enums.py:185]: these were fine but i wanted to make it more clear what they represented

**Review by Johnno1011 — APPROVED** [review]
great stuff thanks for the bomp start - lets getter merged and I'll crack on :D

File: src/primaite/common/enums.py (hunk 1 above)
Comment (Johnno1011) [reply src/primaite/common/enums.py:185]: Got it :D

**Pull request merged**